### PR TITLE
Update `Child` docs to not have a note section

### PR DESCRIPTION
**Pull request by mgattozzi**
In #29370 it's noted that for "the Note shouldn't be one, and should come before
the examples." This commit changes the positioning of the section and removes
wording that said take note in order for it to flow better with the surrounding
text and it's new position.

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -73,6 +73,15 @@ use sys_common::{AsInner, AsInnerMut, FromInner, IntoInner};
 /// spawning process and can itself be constructed using a builder-style
 /// interface.
 ///
+/// There is no implementation of [`Drop`] for child processes,
+/// so if you do not ensure the `Child` has exited then it will continue to
+/// run, even after the `Child` handle to the child process has gone out of
+/// scope.
+///
+/// Calling [`wait`][`wait`] (or other functions that wrap around it) will make
```
Comment (steveklabnik) [inline src/libstd/process.rs]: You don't need to duplicate [`wait`] here, you can drop it, just like [`Drop`] above.

**Review by steveklabnik — CHANGES_REQUESTED** [review]
A very small nit, and then this is good to go. Thanks a ton!

File: src/libstd/process.rs (hunk 1 above)
Comment (mgattozzi) [inline src/libstd/process.rs]: This was already here when I moved it so I had just kept it. I'll drop it.

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thanks! I totally hear you about keeping it the same, might as well fix it while you're here 😄

**Comment by mgattozzi** [comment]
Hahaha yeah I'm not opposed to fixing it, I just wasn't sure if that was some weird markdown thing from before. I figured it'd be better not to touch it.

File: src/libstd/process.rs [hunk 2]
```diff
@@ -73,6 +73,15 @@ use sys_common::{AsInner, AsInnerMut, FromInner, IntoInner};
 /// spawning process and can itself be constructed using a builder-style
 /// interface.
 ///
+/// There is no implementation of [`Drop`] for child processes,
+/// so if you do not ensure the `Child` has exited then it will continue to
+/// run, even after the `Child` handle to the child process has gone out of
+/// scope.
+///
+/// Calling [`wait`] (or other functions that wrap around it) will make
```
Comment (GuillaumeGomez) [inline src/libstd/process.rs]: I'm not sure it'll work. Better write: "[`wait `][ `wait `]"

**Review by GuillaumeGomez — DISMISSED** [review]

**Comment by GuillaumeGomez** [comment]
While waiting for others' confirmation, I'll put this PR on wait. Sorry for the bothering.

@bors: r-

File: src/libstd/process.rs (hunk 2 above)
Comment (steveklabnik) [inline src/libstd/process.rs]: Ah, yes, the space. @mgattozzi did you happen to build this locally? It _should_ work, but it might not.

Comment (mgattozzi) [inline src/libstd/process.rs]: Yeah not a problem. Sorry was a bit busy this week let me quickly edit this

**Comment by mgattozzi** [comment]
This did break the link. Working on the fix now.

**Comment by steveklabnik** [comment]
@mgattozzi let me know when you have this one updated!

**Comment by mgattozzi** [comment]
[ wait ][ wait ] didn't work either. I think we just need to revert the changes. Thoughts @steveklabnik?

**Comment by steveklabnik** [comment]
That's... odd. Given that was the way it was defined below.

**Comment by mgattozzi** [comment]
Hmmm. Looking at it again I had it as  "[ wait ][ wait ]" based off the above comments from @GuillaumeGomez let me retry it without the "" and see if that does it.

**Comment by GuillaumeGomez** [comment]
Yes sure. We're not in a hurry.

**Comment by mgattozzi** [comment]
Yeah that definitely didn't do it. It worked as was before the change I think it might be best if we leave it that way.

**Comment by nagisa** [comment]
```
[`wait`]

[`wait`]: URL
```

Should work just fine if we're following common mark spec properly.

**Comment by GuillaumeGomez** [comment]
@nagisa: Indeed.

**Comment by mgattozzi** [comment]
Alright I'll do that then and see if it works out properly

**Comment by carols10cents** [comment]
Friendly ping to keep this on your radar @mgattozzi!

**Review by steveklabnik — APPROVED** [review]

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thank you!

**Comment by mgattozzi** [comment]
Thanks @carols10cents for the ping. I was a bit busy recently so it was good to get this in.

**Pull request merged**